### PR TITLE
RLP-1014: open channel crash (non-token SC address)

### PR DESCRIPTION
**Pull request by mortelli**
This PR aims to solve [RLP-1014](https://jirainfuy.atlassian.net/browse/RLP-1014), which is about the Lumino node crashing when providing a valid, check-summed Smart Contract address as a Token address when opening a channel.

The solution is to call `total_supply` on the `raiden.network.proxies.Token` object instantiated with the provided address. If an `HTTPError` is raised, the address is assumed not to be a valid Token SC.

It is _possible_ for this call to fail for reasons other than the address not corresponding a Token contract—such as the RSK node throwing a differently-caused `500` at exactly this point—but for now it's a trade-off worth doing. The alternative (which is a crash) is not acceptable in this context.

As usual, _Optimize Imports/Reformat Code_ has been applied to affected files.

### TODO
- [x] manually test solution w/FN
- [x] manually test solution w/LC
- [x] verify `pytest` results

### CODE REVIEW

File: raiden/network/blockchain_service.py [hunk 1]
```diff
@@ -137,14 +139,27 @@ def token(self, token_address: Address) -> Token:
...
                     contract_manager=self.contract_manager,
                 )
 
+                self.check_token_has_supply(token)
+                self.address_to_token[token_address] = token
+
         return self.address_to_token[token_address]
 
+    @staticmethod
+    def check_token_has_supply(token: Token):
```
Comment (mortelli) [inline raiden/network/blockchain_service.py:154]: i wanted to include this function in `raiden/network/rpc/client.py` since it is conceptually similar to `check_address_has_code`. 

the issue was that we don't need a client object to execute this code; i took that as a hint that a better location should be found for it.

i also opted not to have this in the `Token` class. if we were to place it there, it should be in the constructor—we wouldn't want an object which raises an exception on this function—and implementing that means too many changes for it to be worth it atm.

so i think this location is a good middle ground

**Review by JONAF2103 — APPROVED** [review]
LGTM, we can merge after testing

**Comment by mortelli** [comment]
i have ran the `pytest` suite on this branch. my findings are:

1. there are couple more failures compared to `develop`, but barring 1 test, they correspond to known flakes.
2. this new failure is one of the rif comms integration tests. i'm fairly sure that this is _also_ failing on `develop` but we missed it. maybe the `grpc-api` branch was updated between the time the latest results were recorded and now. 
3. there are 8 new errors—`E`, not `F`—but it looks like they all come from tests related to matrix. since the matrix servers had some downtime today, i am going to execute the suite again overnight and see how they turn out.

i'll have more info on points 2 and 3 tomorrow

**Comment by mortelli** [comment]
i have also attempted to manually verify this fix from the LC side, using the [dapp](https://github.com/rsksmart/lumino-lightclient-dapp). 

unfortunately, i am getting notifier errors after onboarding a fresh address.

i am following [the manual testing guide](https://github.com/anarancio/lumino-docs/blob/master/testing/manual/guide.md) and running everything on `master` except for lumino, which is on this branch. i did everything on regtest, from scratch.

@JONAF2103 has kindly agreed to help me test this out tomorrow. it might be the case that an error on the dapp has also slipped by. 

note that we haven't used it in a while; the latest tests were done using [Nifty Wallet Browser Extension](https://github.com/rsksmart/rconnect/).

**Comment by mortelli** [comment]
> i'll have more info on points 2 and 3 tomorrow

so here it is:

>     1. there are couple more failures compared to `develop`, but barring 1 test, they correspond to known flakes.

results that confirm this [have been uploaded](https://github.com/anarancio/lumino-docs/tree/master/testing/pytest/results/bugfix/RLP-1014_non-token-sc-addresses/2021-02-17) to `lumino-docs`.
 
>     2. this new failure is one of the rif comms integration tests. i'm fairly sure that this is _also_ failing on `develop` but we missed it. maybe the `grpc-api` branch was updated between the time the latest results were recorded and now.

i've confirmed this and created an issue to tackle it. see [RLP-1015](https://jirainfuy.atlassian.net/browse/RLP-1015)
 
>     3. there are 8 new errors—`E`, not `F`—but it looks like they all come from tests related to matrix. since the matrix servers had some downtime today, i am going to execute the suite again overnight and see how they turn out.

the errors have not re-occurred, so it seems my theory was correct. this can be verified by taking a look at the results linked above.

in summary: the additional `F`s are either [known flakes](https://github.com/anarancio/lumino-docs/blob/master/testing/pytest/results/develop/2020-12-28/flakes.txt) or [known logging errors](https://jirainfuy.atlassian.net/browse/RLP-799).

i am therefore marking the `pytest` verification as complete

File: raiden/tests/integration/network/transport/rif_comms/test_client.py [hunk 2]
```diff
@@ -466,7 +467,7 @@ def test_unsubscribe_from_invalid_address(comms_clients):
     # bypass client.unsubscribe_from to provide an invalid address as topic
     expect_error(
         expected_exception=RpcError,
-        expected_mapped_type=NotFoundException,
+        expected_mapped_type=InvalidArgumentException,
```
Comment (mortelli) [inline raiden/tests/integration/network/transport/rif_comms/test_client.py:470]: piggybacked correction here (also on line `481`)

File: raiden/tests/integration/network/transport/rif_comms/test_client.py [hunk 3]
```diff
@@ -456,6 +456,7 @@ def test_unsubscribe_from_non_subscribed_address(comms_clients):
 
 
 @pytest.mark.parametrize("nodes_to_clients", [{"A": 1}])
+@pytest.mark.xfail(reason="wrong error code/msg from comms")
```
Comment (mortelli) [inline raiden/tests/integration/network/transport/rif_comms/test_client.py:459]: more info in [RLP-1015](https://jirainfuy.atlassian.net/browse/RLP-1015)

**Comment by mortelli** [comment]
> i have also attempted to manually verify this fix from the LC side, using the [dapp](https://github.com/rsksmart/lumino-lightclient-dapp).
> 
> unfortunately, i am getting notifier errors after onboarding a fresh address.
> 
> i am following [the manual testing guide](https://github.com/anarancio/lumino-docs/blob/master/testing/manual/guide.md) and running everything on `master` except for lumino, which is on this branch. i did everything on regtest, from scratch.
> 
> @JONAF2103 has kindly agreed to help me test this out tomorrow. it might be the case that an error on the dapp has also slipped by.
> 
> note that we haven't used it in a while; the latest tests were done using [Nifty Wallet Browser Extension](https://github.com/rsksmart/rconnect/).

i got this to work (thanks @JONAF2103 for your help with it).

it seems attempting to supply a non-token smart contract address as the token address when opening a channel in the dapp results in the following HUB error:

```
[error    ] Error processing request       [raiden.api.rest] errors=No token network registered for token "0x45aA64F4A3A4BaCb863B6235968c9fFBbd6D4778" status_code=<HTTPStatus.NOT_FOUND: 404>
```

i am therefore marking the LC verification as complete and merging
